phoenix-core/src/it/java/org/apache/phoenix/end2end/SystemCatalogCreationOnConnectionIT.java

```diff
@@ -171,6 +173,7 @@ public void tearDownMiniCluster() {
 
      // Conditions: isDoNotUpgradePropSet is true
      // Expected: We do not create SYSTEM.CATALOG even if this is the first connection to the server
+     // UpgradeRequiredException is also not thrown even if it is first connection to server
```

Comment:
Based on my comments below, this test can probably stay unchanged if we catch TNFE and rethrow as UpgradeRequiredException.